### PR TITLE
added api integration with dapr

### PR DESCRIPTION
**Pull request by manojNeosoft**

### CODE REVIEW

File: src/components/factory/FactoryController.ts [hunk 1]
```diff
@@ -1,30 +1,87 @@
...
         } catch (error) {
             throw (error)
         }
     }
+
+    public async getStore(params) {
+        try {
+            const daprPort = process.env.DAPR_HTTP_PORT || 3500;
+            const stateStoreName = `statestore`;
+            const stateUrl = `http://localhost:${daprPort}/v1.0/state/${stateStoreName}`;
```
Comment (sagarj-webmaster) [inline src/components/factory/FactoryController.ts]: Take URL from env or constant file

File: src/components/factory/FactoryController.ts [hunk 2]
```diff
@@ -1,30 +1,87 @@
...
+            }
         } catch (error) {
             throw (error)
         }
     }
+
+    public async getStore(params) {
+        try {
+            const daprPort = process.env.DAPR_HTTP_PORT || 3500;
+            const stateStoreName = `statestore`;
```
Comment (sagarj-webmaster) [inline src/components/factory/FactoryController.ts]: stateStoreName  take from constant

File: src/components/factory/FactoryController.ts [hunk 3]
```diff
@@ -1,30 +1,87 @@
...
+                headers: { "Content-Type": "application/json" }
+            })
+            if(result.data) {
+                console.log("State saved.")
+                return { data: { status: 200, message: "Successfully persisted state" } };
+            } else {
+                return result;
+            }
         } catch (error) {
             throw (error)
```
Comment (sagarj-webmaster) [inline src/components/factory/FactoryController.ts:43]: catch use common error handle function  ResponseHandler.JSONERROR(request, response);

File: src/components/factory/FactoryController.ts [hunk 4]
```diff
@@ -1,30 +1,87 @@
...
+    public async getFactory(params) {
         try {
-            const result: IFatctoryInputModel[] = await this.factoryRepository.find()
-            return {
-                data : result
-            };
+            // const result: IFatctoryInputModel[] = await this.factoryRepository.find()
+            // return {
+            //     data: result
+            // };
```
Comment (sagarj-webmaster) [inline src/components/factory/FactoryController.ts]: remove commented code

File: src/components/factory/FactoryController.ts [hunk 5]
```diff
@@ -1,30 +1,87 @@
...
+            const stateStoreName = `statestore`;
+            const stateUrl = `http://localhost:${daprPort}/v1.0/state/${stateStoreName}`;
+            const result = await axios({
+                url: `${stateUrl}/${params.id}`,
+                method: "GET",
+                params: {},
+                headers: { "Content-Type": "application/json" }
+            })
+            if(result.data) {
+                    console.log("Successfully retrieved persisted state.")
```
Comment (sagarj-webmaster) [inline src/components/factory/FactoryController.ts]: remove log

File: src/components/factory/FactoryController.ts (hunk 2 above)
Comment (manojNeosoft) [reply src/components/factory/FactoryController.ts]: done

File: src/components/factory/FactoryController.ts (hunk 5 above)
Comment (manojNeosoft) [reply src/components/factory/FactoryController.ts]: done

File: src/components/factory/FactoryController.ts (hunk 4 above)
Comment (manojNeosoft) [reply src/components/factory/FactoryController.ts]: done

**Pull request merged**